### PR TITLE
DCOS-21844: fix webpack hot-loader

### PR DESCRIPTION
**Pull request by bstavroulakis**
Fix hot-loader.
~Improve inline-bootstrap-css.~ => https://github.com/dcos/dcos-ui/pull/2866
~Remove open:true.~ => https://github.com/dcos/dcos-ui/pull/2867

**Checklist**
- [ ] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

File: src/js/index.js [hunk 1]
```diff
@@ -63,7 +64,9 @@ RequestUtil.json = function(options = {}) {
 (function() {
   function renderApplication() {
     function renderAppToDOM(content) {
-      ReactDOM.render(content, domElement, function() {
+      console.log(content);
+      const hotContent = hot(module)(() => content);
```
Comment (DanielMSchmidt) [inline src/js/index.js]: Does this work? I tried something similar while approaching this problem in the first place, but it didn't work for me

File: webpack/webpack.dev.js [hunk 2]
```diff
@@ -8,11 +8,21 @@ const dependencies = Object.assign({}, packageInfo.dependencies);
 delete dependencies["canvas-ui"];
 delete dependencies["cnvs"];
 
+const entry = {
+  index: ["./src/js/index.js"],
+  vendor: Object.keys(dependencies)
+};
+
+// if (process.env.NODE_ENV === "development") {
+//   entry.index.unshift(
```
Comment (DanielMSchmidt) [inline webpack/webpack.dev.js]: I know it's commented out, but let' not doo so much magic here

File: webpack/webpack.config.js [hunk 3]
```diff
@@ -44,8 +44,9 @@ module.exports = {
   },
   devServer: {
     // TODO: https://webpack.js.org/configuration/dev-server/#devserver-hot
+    hot: true,
     contentBase: path.join(__dirname, "../dist"),
-    open: true,
+    open: false,
```
Comment (juliangieseke) [inline webpack/webpack.config.js]: Ok 👍 (I'm indifferent about this…)

File: webpack/webpack.config.js [hunk 4]
```diff
@@ -83,7 +84,8 @@ module.exports = {
               "babel-preset-es2015",
               "babel-preset-stage-3",
               "babel-preset-react"
-            ])
+            ]),
+            plugins: ["react-hot-loader/babel"]
```
Comment (juliangieseke) [inline webpack/webpack.config.js]: again, dev config only?

**Comment by juliangieseke** [comment]
Hey Bill, I have extracted the two already finished parts from this PR: 
https://github.com/dcos/dcos-ui/pull/2866 
https://github.com/dcos/dcos-ui/pull/2867

would be nice if you can quickly clean up this one 👍

**Comment by juliangieseke** [comment]
nvm, cleaned up for you 👍 hope that was okay 🙇 
old sha if you need to recover something: `ff4cc60a5e94966c408025abed37eadf8e33049d`

**Comment by juliangieseke** [comment]
shall we close this? 😢

**Pull request closed without merging**